### PR TITLE
Port keras magic

### PR DESCRIPTION
**Pull request by farizrahman4u**

### CODE REVIEW

File: wandb/integration/keras/magic_impl.py [hunk 1]
```diff
@@ -0,0 +1,571 @@
+from __future__ import absolute_import
```
Comment (vanpelt) [inline wandb/integration/keras/magic_impl.py:1]: @raubitsj may have some thoughts on where to put this.  I don't think it belongs inside of `keras`, the magic implementation was designed to support non-keras usecases as well.  Maybe `integration/magic`?

Comment (raubitsj) [reply wandb/integration/keras/magic_impl.py:1]: integration/magic sounds good for now.

Comment (farizrahman4u) [reply wandb/integration/keras/magic_impl.py:1]: On doing this refactor, (`integration/keras/magic_impl.py` ->`integration/magic.py`)  the tfkreras patching for getting validation data fails. Looking into it now. Probably something to do with import order.

Comment (farizrahman4u) [reply wandb/integration/keras/magic_impl.py]: Fixed.

File: wandb/integration/keras/keras.py [hunk 2]
```diff
@@ -64,6 +47,10 @@ def is_generator_like(data):
         data, types))
 
 
+def patch_keras():
+    # TODO
+    pass
```
Comment (vanpelt) [inline wandb/integration/keras/keras.py:52]: What's intended here?  I believe regular keras doesn't need to be patched.

File: wandb/sdk/wandb_run.py [hunk 3]
```diff
@@ -41,7 +41,7 @@
 
 
 logger = logging.getLogger("wandb")
-EXIT_TIMEOUT = 60
+EXIT_TIMEOUT = 10000000000
```
Comment (vanpelt) [inline wandb/sdk/wandb_run.py:44]: What's up here?

Comment (farizrahman4u) [reply wandb/sdk/wandb_run.py:44]: Wasn't supposed to check it in.

File: wandb/integration/keras/keras.py (hunk 2 above)
Comment (farizrahman4u) [reply wandb/integration/keras/keras.py:52]: Will remove after testing against regular keras.

Comment (farizrahman4u) [reply wandb/integration/keras/keras.py:52]: Nope. Getting `wandb: WARNING No validation_data set, pass a generator to the callback.` with regular keras. Will take a look tomorrow.

Comment (farizrahman4u) [reply wandb/integration/keras/keras.py]: Fixed

File: wandb/sdk/wandb_config.py [hunk 4]
```diff
@@ -94,6 +94,9 @@ def __setitem__(self, key, val):
         if self._callback:
             self._callback(key=key, val=val, data=self._as_dict())
 
+    def items(self):
+        return self._items.items()
```
Comment (raubitsj) [inline wandb/sdk/wandb_config.py:98]: I think we might want to hide some items.  namely anything with the key `_wandb`, but it looks like elsewhere we are hiding anything beginning with "_"

File: wandb/sdk/wandb_init.py [hunk 5]
```diff
@@ -13,6 +13,7 @@
```
Comment (raubitsj) [inline wandb/sdk/wandb_init.py:13]: Need to handle 
wandb.init(magic=True)
and possibly the extended syntax of
wandb.init(magic=dict(...))

File: wandb/integration/keras/keras.py [hunk 6]
```diff
@@ -147,12 +130,10 @@ def new_v2(*args, **kwargs):
...
-if "tensorflow" in wandb.util.get_full_typename(keras):
-    try:
-        patch_tf_keras()
-    except Exception:
-        wandb.termwarn(
-            "Unable to patch tensorflow.keras for use with W&B.  You will not be able to log images unless you set the generator argument of the callback.")
+
+import tensorflow.keras as keras
+import tensorflow.keras.backend as K
+patch_tf_keras()
```
[the remark this answers is not in the document]
Comment (farizrahman4u) [reply wandb/integration/keras/keras.py:136]: This works with standalone keras as well, as Keras 2.4 is simply a pointer to tf.keras: https://github.com/keras-team/keras/releases

[the remark this answers is not in the document]
Comment (vanpelt) [reply wandb/integration/keras/keras.py:136]: Cool, I would still be concerned about earlier versions of keras.

[the remark this answers is not in the document]
Comment (farizrahman4u) [reply wandb/integration/keras/keras.py:150]: Added warning for old version of keras

File: wandb/integration/keras/keras.py [hunk 7]
```diff
@@ -544,6 +557,9 @@ def _log_images(self, num_images=36):
             if self.output_type == "label":
                 # we just use the predicted label as the caption for now
                 captions = self._logits_to_captions(predictions)
+                print(len(captions))
+                print(captions)
+                print(len(test_data))
```
Comment (raubitsj) [inline wandb/integration/keras/keras.py:562]: remove debugging messages

**Pull request merged**